src/pages/CreateBillableMetric.tsx

```diff
@@ -129,13 +156,33 @@ const CreateBillableMetric = () => {
     }),
     enableReinitialize: true,
     validateOnMount: true,
-    onSubmit: onSave,
+    onSubmit: (values) => {
+      if (values.aggregateOnTab === AggregateOnTab.CustomExpression) {
+        return onSave(_omit(values, ['aggregateOnTab']))
```

Comment:
Refactor suggestion: 
Have an approach "if this, then modify values" then have a simple return given the formated values at the end.
Why? To make is more predictable and easier to encapsulate in a serialiser method we can extract and save later if the code starts to have more complexity.
The would lead to something like
```
const formatedValues = _omit(values, ['aggregateOnTab'])
if (values.aggregateOnTab !== AggregateOnTab.CustomExpression)
  // and maybe add a comment about expression only being used for custom expressions
  formatedValues.expression = null

return onSave(formatedValues)
```


Just a suggestion let me know of that make sense!

Reply:
Interesting!

First, in my opinion, the idea of comments describing code isn't optimal. The effects and the reasoning behind our code should be understandable without comments. If we rely on comments to help developers understand the flow, we introduce technical debt since those comments have to be maintained.

Secondly, mutating values in the middle of functions could lead to [unwanted side-effects](https://sapegin.me/blog/avoid-mutation). Your solution, in the context of the current function, doesn't seem troublesome since the function is small. But if we keep mutations like that as a pattern in our codebase, functions that are bigger could become hard-to-follow / hard-to-debug. Indeed, the `onSubmit` could benefit from some refactoring, but it should be done in an immutable way (extract different branches to functions). If you think we should do it now, we could, but I would say that it's a bit of an early optimization.

Thanks!

Reply:
up to you!